src/coreComponents/common/format/table/TableFormatter.cpp

```diff
@@ -127,284 +164,466 @@ TableTextFormatter::TableTextFormatter( TableLayout const & tableLayout ):
 
-  outputSectionRows( columns, sectionSeparatingLine, tableOutput, nbHeaderRows, TableLayout::Section::header );
+  outputHeaderSectionRows( tableColumnsData, tableOutput, nbHeaderRows, sectionSeparatingLine );
+
+  outputValuesSectionRows( tableColumnsData, tableOutput, nbValuesRows, sectionSeparatingLine );
+  if( m_tableLayout.isLineBreakEnabled())
+  {
+    tableOutput << '\n';
+  }
+}
+
+void TableTextFormatter::populateColumnsFromTableData( std::vector< TableLayout::TableColumnData > & tableColumnsData,
+                                                       std::vector< std::vector< string > > const & tableDataRows,
+                                                       bool isSubColumn ) const
+{
+  size_t currentColumn = 0;
+  std::vector< std::vector< string > > tColumnsValues( tableDataRows[0].size(),
+                                                       std::vector< string >( tableDataRows.size()));
+  if( !isSubColumn )
+  {
+    transpose( tColumnsValues, tableDataRows );
+  }
+
+  for( auto & tableColumnData : tableColumnsData )
+  {
+    if( tableColumnData.subColumn.empty())
+    {
+      tableColumnData.columnValues = !isSubColumn ?
+                                     tColumnsValues[currentColumn++] : tableDataRows[currentColumn++];
+    }
+    else
+    {
+      size_t nbSubColumns = tableColumnData.column.subColumnNames.size();
+      auto subColumnStartIdx = tColumnsValues.begin() + currentColumn;
+      std::vector< std::vector< string > > subColumnValues( subColumnStartIdx,
+                                                            subColumnStartIdx + nbSubColumns );
+      populateColumnsFromTableData( tableColumnData.subColumn, subColumnValues, true );
+
+      std::vector< std::vector< string > > tSubColumnValues( subColumnValues[0].size(),
+                                                             std::vector< string >( subColumnValues.size()) );
+      transpose( tSubColumnValues, subColumnValues );
+      for( const auto & columnValues : tSubColumnValues )
+      { // add all subcolumn values in parent tableColumnData
+        tableColumnData.columnValues.insert( tableColumnData.columnValues.end(),
+                                             columnValues.begin(),
+                                             columnValues.end() );
+      }
+      currentColumn += subColumnValues.size();
+    }
+  }
 }
 
-void TableTextFormatter::splitAndSetColumnNames( std::vector< TableLayout::Column > & columns,
-                                                 size_t & nbHeaderRows,
-                                                 std::vector< std::vector< string > > & splitHeaders ) const
+void TableTextFormatter::splitAndMergeColumnHeaders( std::vector< TableLayout::TableColumnData > & tableColumnsData,
+                                                     size_t & nbHeaderRows,
+                                                     std::vector< std::vector< string > > & splitHeaders ) const
```

Comment:
I removed nbHeaderRows parameters in all signature, no issue for we add '\n' to subcolumn bc it's not implemented yet, it will just take the first part of the string

Reply:
I think it needs more work to either fully implement what those lines tries to do:
```c++
    if( !tableColumnData.subColumn.empty())
    {
      std::vector< std::vector< string > > splitSubColHeaders;
      size_t nbHeaderSubColRows = 0;
      splitAndMergeColumnHeaders( tableColumnData.subColumn, nbHeaderSubColRows, splitSubColHeaders );
    }
```